plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSinkProvider.java

```diff
@@ -193,7 +193,7 @@ private DeltaLakeCdfPageSink createCdfPageSink(
     {
         MetadataEntry metadataEntry = mergeTableHandle.getTableHandle().getMetadataEntry();
         Set<String> partitionKeys = mergeTableHandle.getTableHandle().getMetadataEntry().getOriginalPartitionColumns().stream().collect(toImmutableSet());
-        List<DeltaLakeColumnHandle> tableColumns = extractSchema(metadataEntry, typeManager).stream()
+        List<DeltaLakeColumnHandle> tableColumns = extractSchema(metadataEntry, mergeTableHandle.getTableHandle().getProtocolEntry(), typeManager).stream()
```

Comment:
Extract `mergeTableHandle.getTableHandle().getProtocolEntry()` to a variable because it is being used several times within this method.